### PR TITLE
fix(PropertyExt): Improved changed hashing

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

File: packages/common/src/PropertyExt.ts [hunk 1]
```diff
@@ -682,10 +682,21 @@ export class PropertyExt extends Class {
...
         this.publishedProperties(false).filter(meta => ignore.indexOf(meta.id) < 0).forEach(meta => {
             if (this[meta.id + "_exists"]()) {
+                let value = this[meta.id]();
+                switch (meta.type) {
+                    case "widget":
+                        value = value.hashSum();
+                        break;
+                    case "widgetArray":
+                    case "PropertyArray":
+                        value = hashSum(value.map(v => v.hashSum()));
```
Comment (jbrundage) [inline packages/common/src/PropertyExt.ts]: It might be a good idea to rename the imported `hashSum` to `utilityHashSum` to avoid confusion?

Comment (jbrundage) [reply packages/common/src/PropertyExt.ts]: `value` seems to be unused in this context? Does assigning the response from `hashSum` to it somehow affect `this[meta.id]` as a pass by reference?

Comment (GordonSmith) [reply packages/common/src/PropertyExt.ts]: It is used further down the source code (just not in the changed code).

Comment (GordonSmith) [reply packages/common/src/PropertyExt.ts]: Hmmm it does look iffy.

**Comment by GordonSmith** [comment]
The idea is it should improve performance for the array of object "widgets" as currently it was calculating the hash on the entire "class" instance (including functions etc.).

**Comment by GordonSmith** [comment]
(...fixed now uses new value)

**Pull request merged**